docs/getting-started/custom-ui-preact-ory-elements.mdx

```diff
@@ -0,0 +1,226 @@
+provided by Ory Elements to interact with Ory Network APIs to experience real-life user interactions and the dynamic nature of
+Elements.
+
+To install both dependencies, run this command:
+
+```shell-session
+npm i @ory/client @ory/elements-preact wouter --save
+```
+
+After installing the dependencies, your `package.json` should look similar to this:
+
+```mdx-code-block
+<CodeFromRemote
+  lang="json"
+  title="package.json"
+  src="https://raw.githubusercontent.com/ory/elements/main/examples/preact-spa/package.json"
+/>
+```
+
+## Integrate Ory Elements
+
+After creating the application and installing dependencies, you can start integrating Ory Elements and Ory SDK.
+
+### Create new instance of Ory SDK
+
+The `sdk.ts` file is responsible for creating a new instance of the Ory SDK. This instance will be used to interact with Ory APIs.
+
+```mdx-code-block
+<CodeFromRemote
+  lang="ts"
+  title="src/sdk.ts"
+  src="https://raw.githubusercontent.com/ory/elements/main/examples/preact-spa/src/sdk.ts"
+/>
+```
+
+### Create Preact entry point
+
+The `main.tsx` file is the main entry point for Preact. This file contains routes and themes for other pages used by the app. The
+application is wrapped with Ory Elements using `<ThemeProvider />`.
+
+```mdx-code-block
+<CodeFromRemote
+  lang="tsx"
+  title="src/main.tsx"
+  src="https://raw.githubusercontent.com/ory/elements/main/examples/preact-spa/src/main.tsx"
+/>
+```
+
+### Create landing page
+
+The `app.tsx` file defines the landing page of the application. If you want to protect access to an app, set up authentication to
+secure this page.
+
+```mdx-code-block
+<CodeFromRemote
+  lang="tsx"
+  title="src/app.tsx"
+  src="https://raw.githubusercontent.com/ory/elements/main/examples/preact-spa/src/app.tsx"
+/>
+```
+
+### Login and registration pages
+
+The `login.tsx` and `register.tsx` pages contain components that interact with Ory APIs to allow registering accounts and
+performing the login flow.
+
+These pages are dynamic and will show relevant fields based on your Ory Network project configuration. To see the pages change,
+edit your project's [Identity Schema](../kratos/manage-identities/identity-schema) or try changing adjusting project settings and
+enable [passwordless](../kratos/passwordless/webauthn) flows.
+
+#### Registration
+
+The registration page, defined in the `register.tsx` file:
+
+1. Gets the registration flow (if it exists in the URL) or fallback to start the registration flow to get the `flow` data on page
+   load.
+2. Submits the registration form data the user provides when they click the **Submit** button,
+
+:::tip
+
+To learn more about the self-service registration flow, read [this document](../kratos/self-service/flows/user-registration)
+
+:::
+
+```mdx-code-block
+<CodeFromRemote
+  lang="tsx"
+  title="src/register.tsx"
+  src="https://raw.githubusercontent.com/ory/elements/main/examples/preact-spa/src/register.tsx"
```

Comment:
It probably makes sense to put these fetching mechanisms into a React `useEffect`? WDYT @mszekiel ?

Not blocking the merge of course!

Reply:
Yes, this section of `onSubmit` could be moved to a handler function above 

```ts
onSubmit={({ body }) => {
        sdk
          .submitSelfServiceRegistrationFlow(
            flow.id,
            body as SubmitSelfServiceRegistrationFlowBody,
          )
          .then(() => {
            // we successfully submitted the login flow, so lets redirect to the dashboard
            setLocation("/", { replace: true })
          })
          .catch((error) => {
            setFlow(error.response.data)
          })
      }}
      ```

Reply:
Also in different parts, mentioned that in PM with Tom

Reply:
Since I'm merely showing code from another repo, these improvements require updating the code of the example that lives here: 
https://github.com/ory/elements/tree/main/examples/preact-spa

;)

Reply:
Yeah yeah absolutely, it's not related to this PR but generally ory elements